engine/src/cmd/carrier.cpp

```diff
@@ -61,10 +61,10 @@ static int stoi(const string &inp, int def = 0) {
 class CatCompare {
 public:
     bool operator()(const Cargo &a, const Cargo &b) {
-        std::string::const_iterator aiter = a.GetCategory().begin();
-        std::string::const_iterator aend = a.GetCategory().end();
-        std::string::const_iterator biter = b.GetCategory().begin();
-        std::string::const_iterator bend = b.GetCategory().end();
+        std::string::const_iterator aiter = a.category.begin();
```

Comment:
This is where the `GetCategory()` can show a difference; this would make more sense to me as `GetCategories()` where `GetCategory()` returns a single or raw value, and `GetCategories()` returns a `std::list` (or `std::deque`) of all the categories that could be encoded within the category data since the use of the iterator seems to imply we're splitting the string data up for some reason.

Similarly a `HasCategory()` could be implemented so that `c.category.find("upgrades")` would be `c.HasCategory("upgrades")` instead where `HasCategory(name)` would be implemented like:
```
categories = c.GetCategories()
if name in categories return true
else return false
```

Reply:
Maybe later. Changing this and back has been expensive time-wise. I'll refactor this when I revisit the class.